code/modules/reagents/Chemistry-Machinery.dm

```diff
@@ -809,3 +809,17 @@
 			target.say("*scream")
 			spawn(10)
 			user.visible_message("<span class='warning'>[user] stops the [src] and leaves [target] resting as they are.</span>", "<span class='warning'>You turn the [src] off and let go of [target].</span>")
+
+/obj/machinery/reagentgrinder/verb/Eject()
+	set src in oview(1)
+	set category = "Object"
+	set name = "Eject contents"
+
+	usr.visible_message(
+	"<span class='notice'>[usr] open's [src] and has removed [english_list(contents)].</span>"

```

Comment:
I do not think that a visible message is needed here.

Reply:
Not really the best or the worst, but:
* Fix the grammar a little -> `[usr] opens [src] and has removed [english_list(holdingitems)]`
* Do this text _after_ the `do_after` check.